xla/service/gpu/fusions/mlir/mlir_fusion_emitter.cc

```diff
@@ -590,8 +593,25 @@ void AddLoweringPasses(mlir::OpPassManager& pm,
   pm.addPass(mlir::createLoopInvariantCodeMotionPass());
   pm.addPass(mlir::createSymbolDCEPass());
   pm.addPass(mlir::createCSEPass());
-  pm.addPass(CreateExpandFloatOpsPass(
-      !device.cuda_compute_capability().IsAtLeastAmpere()));
+
+#ifdef GOOGLE_CUDA
+  // The call to `GetAsmCompilerVersion` is expensive, but the result never
```

Comment:
Can this check be moved to the pass?
Currently, we get a build breakage due to the header include of xla/stream_executor/cuda/cuda_asm_compiler.h
This could be fixed by yet another #ifdef, but we try to avoid those. So if the check is moved to a pass that is compiled only for cuda, the include can be used unconditionally.

Reply:
Do we have an example of passes that are compiled for a specific platform only?

Will I need to create a separate "passes_nvidia.td" file (along with the header)?

Reply:
Unfortunately no examples yet, and thinking a bit more about this, I can understand the problem and why you ask about separate passes_nvidia.td file. If we have a pass that should be compiled on a specific platform only, we need to create a stub implementation if it is compiled on a different platform. So at least on the BUILD side I cannot see a way to avoid a if_cuda_is_configured guard, but we can avoid the #ifdef in the code.
So essentially my suggestion would be to create a convert_float_nvidia_stub.cc file that returns a pass that does nothing, and use the cuda_asm_compiler.h header only in the "real" convert_float_nvidia.cc file. Then, we need neither a #ifdef for the header include, nor for adding the pass to the pipeline.

Reply:
Updated the code, looks cleaner. PTAL